tests/unit-test-server.mjs

```diff
@@ -0,0 +1,50 @@
+// Simple server adapted from https://developer.mozilla.org/en-US/docs/Learn/Server-side/Node_server_without_framework:
```

Comment:
Nit: I assume we'll likely continue to use our testing setup even when we introduce larger end-to-end tests? If that's the case, maybe we can just rename these files to `run.mjs` and `server.mjs`?

Reply:
Yeah, we may want to pull in some dependencies if this gets more complex but the generic names make sense.